# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/promiseworker/worker/PromiseWorker.js
# Commit: 5ea572e91652
# Full Hash: 5ea572e916522c6a1a96daef9f37427ab245ed57
# Author: Emma Malysz <emalysz@mozilla.com>
# Date: 2021-05-11 02:09:31
# Regressor Bug: 1649602
# File Overlap Count: 2
# Description:
#   Bug 1649602, remove OS.File from PageThumbs.jsm r=Gijs,barret
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101066
# ==============================================================================

diff -r a00207a14a8b -r 5ea572e91652 toolkit/components/promiseworker/worker/PromiseWorker.js
--- a/toolkit/components/promiseworker/worker/PromiseWorker.js	Mon May 10 16:06:34 2021 +0000
+++ b/toolkit/components/promiseworker/worker/PromiseWorker.js	Mon May 10 16:18:44 2021 +0000
@@ -172,6 +172,16 @@
       } else {
         this.postMessage({ ok: result, id, durationMs });
       }
+    } else if (exn.constructor.name == "DOMException") {
+      // We can receive instances of DOMExceptions with file I/O.
+      // DOMExceptions are not yet serializable (Bug 1561357) and must be
+      // handled differently, as they only have a name and message
+      this.log("Sending back DOM exception", exn.constructor.name);
+      let error = {
+        exn: exn.constructor.name,
+        message: exn.message,
+      };
+      this.postMessage({ fail: error, id, durationMs });
     } else if (exn.constructor.name in EXCEPTION_NAMES) {
       // Rather than letting the DOM mechanism [de]serialize built-in
       // JS errors, which loses lots of information (in particular,